tests/Unit/Domain/DomainCreators/Test_Brick.cpp

```diff
@@ -0,0 +1,196 @@
+           {Direction<3>::lower_eta(), {0, aligned_orientation}},
+           {Direction<3>::upper_eta(), {0, aligned_orientation}}}},
+      std::vector<std::unordered_set<Direction<3>>>{
+          {{Direction<3>::lower_zeta()}, {Direction<3>::upper_zeta()}}});
+
+  const DomainCreators::Brick periodic_yz_brick{
+      lower_bound, upper_bound, std::array<bool, 3>{{false, true, true}},
+      num_elements, grid_points};
+  test_brick_construction(
+      periodic_yz_brick, lower_bound, upper_bound, grid_points, num_elements,
+      std::vector<std::unordered_map<Direction<3>, BlockNeighbor<3>>>{
+          {{Direction<3>::lower_eta(), {0, aligned_orientation}},
+           {Direction<3>::upper_eta(), {0, aligned_orientation}},
+           {Direction<3>::lower_zeta(), {0, aligned_orientation}},
+           {Direction<3>::upper_zeta(), {0, aligned_orientation}}}},
+      std::vector<std::unordered_set<Direction<3>>>{{
+          {Direction<3>::lower_xi()},
+          {Direction<3>::upper_xi()},
+      }});
+
+  const DomainCreators::Brick periodic_xz_brick{
+      lower_bound, upper_bound, std::array<bool, 3>{{true, false, true}},
+      num_elements, grid_points};
+  test_brick_construction(
+      periodic_xz_brick, lower_bound, upper_bound, grid_points, num_elements,
+      std::vector<std::unordered_map<Direction<3>, BlockNeighbor<3>>>{
+          {{Direction<3>::lower_xi(), {0, aligned_orientation}},
+           {Direction<3>::upper_xi(), {0, aligned_orientation}},
+           {Direction<3>::lower_zeta(), {0, aligned_orientation}},
+           {Direction<3>::upper_zeta(), {0, aligned_orientation}}}},
+      std::vector<std::unordered_set<Direction<3>>>{
+          {{Direction<3>::lower_eta()}, {Direction<3>::upper_eta()}}});
+
+  const DomainCreators::Brick periodic_xyz_brick{
+      lower_bound, upper_bound, std::array<bool, 3>{{true, true, true}},
+      num_elements, grid_points};
+  test_brick_construction(
+      periodic_xyz_brick, lower_bound, upper_bound, grid_points, num_elements,
+      std::vector<std::unordered_map<Direction<3>, BlockNeighbor<3>>>{
+          {{Direction<3>::lower_xi(), {0, aligned_orientation}},
+           {Direction<3>::upper_xi(), {0, aligned_orientation}},
+           {Direction<3>::lower_eta(), {0, aligned_orientation}},
+           {Direction<3>::upper_eta(), {0, aligned_orientation}},
+           {Direction<3>::lower_zeta(), {0, aligned_orientation}},
+           {Direction<3>::upper_zeta(), {0, aligned_orientation}}}},
+      std::vector<std::unordered_set<Direction<3>>>{{}});
+}
+
+SPECTRE_TEST_CASE("Unit.Domain.DomainCreators.Brick.Factory",
+                  "[Domain][Unit]") {
+  test_factory_creation<DomainCreator<3, Frame::Inertial>>(
+      "  Brick:\n"
+      "    LowerBound: [0,0,0]\n"
+      "    UpperBound: [1,2,3]\n"
+      "    IsPeriodicIn: [True,False,True]\n"
+      "    InitialGridPoints: [3,4,3]\n"
+      "    InitialRefinement: [2,3,2]\n");
+}
+
+// [[OutputRegex, index = 1]]
```

Comment:
doesn't the ASSERT print more than this? i.e. I would expect it to print the condition that failed as well. If so, you should check that as part of the regex. The same applies for the test below, and in rectangle and interval.

Reply:
Changed my mind, function will disappear anyway.